test/suite/test_turtle01.py

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python
+#
+# Public Domain 2014-present MongoDB, Inc.
+# Public Domain 2008-2014 WiredTiger, Inc.
+#
+# This is free and unencumbered software released into the public domain.
+#
+# Anyone is free to copy, modify, publish, use, compile, sell, or
+# distribute this software, either in source code form or as a compiled
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+import wiredtiger
+import wttest
+import re
+
+# test_turtle.py
+# The following test is to validate the turtle file and to ensure it
+# contains the correct key-value pairs.
+
+
+class test_turtle01(wttest.WiredTigerTestCase):
+    uri = "table:test_turtle"
+    nrows = 1000
+    WT_METADATA_VERSION = "WiredTiger version"
+    WT_METADATA_VERSION_FORMAT = "major=(\d+),minor=(\d+),patch=(\d+)"
+    WT_METADATA_VERSION_STRING = "WiredTiger version string"
+    WT_METADATA_VERSION_STRING_FORMAT = "(\d+)\.(\d+)\.(\d+)"
+    WT_TURTLE_FILE_NAME = "WiredTiger.turtle"
+
+    def init_values(self):
+        self.val = "aaaa"
+        self.key_format = "i"
+        self.value_format = "S"
+
+    def test_validate_turtle_file(self):
+
+        # Validate the .turtle file for the empty database.
+        self.read_turtle()
+        self.check_turtle()
+
+        # Create a table and perform some I/O.
+        self.init_values()
+        create_params = "key_format={},value_format={}".format(
+            self.key_format, self.value_format)
+
+        self.session.create(self.uri, create_params)
+        cursor = self.session.open_cursor(self.uri)
+
+        self.session.begin_transaction()
+        for i in range(1, self.nrows + 1):
+            cursor[i] = self.val
+        self.session.commit_transaction()
+        self.session.checkpoint()
```

Comment:
After a checkpoint, there should be a metadata checkpoint info in the turtle file. We need to verify that as well.

Reply:
I added a function on line 78 to check that the metadata line is comma-separated. Can you review again?